node/src/manager/commands/index.rs

```diff
@@ -44,13 +55,130 @@ pub async fn list(
+        writeln!(term, "{:-^76}", "")?;
+        Ok(())
+    }
+
+    fn footer(term: &mut Terminal) -> Result<(), anyhow::Error> {
+        writeln!(term, "  (a): account-like flag set")?;
+        Ok(())
+    }
+
+    fn print_index(term: &mut Terminal, index: &CreateIndex) -> CmdResult {
+        use CreateIndex::*;
+
+        match index {
+            Unknown { defn } => {
+                writeln!(term, "*unknown*")?;
+                writeln!(term, "  {defn}")?;
+            }
+            Parsed {
+                unique,
+                name,
+                nsp: _,
+                table: _,
+                method,
+                columns,
+                cond,
+                with,
+            } => {
+                let unique = if *unique { " unique" } else { "" };
+                let start = format!("{unique} using {method}");
+                let columns = columns.into_iter().map(|c| c.to_string()).join(", ");
+
+                term.green()?;
+                if index.is_default_index() {
+                    term.dim()?;
+                } else {
+                    term.bold()?;
+                }
+                write!(term, "{name}")?;
+                term.reset()?;
+                write!(term, "{start}")?;
+                term.blue()?;
+                if name.len() + start.len() + columns.len() <= 76 {
+                    writeln!(term, "({columns})")?;
+                } else {
+                    writeln!(term, "\n  on ({})", columns)?;
+                }
+                term.reset()?;
+                if let Some(cond) = cond {
+                    writeln!(term, "  where {cond}")?;
+                }
+                if let Some(with) = with {
+                    writeln!(term, "  with {with}")?;
+                }
+            }
+        }
+        Ok(())
+    }
+
     let deployment_locator = search.locate_unique(&pool)?;
-    let indexes: Vec<String> = store
-        .indexes_for_entity(&deployment_locator, entity_name)
-        .await?;
-    for index in &indexes {
-        println!("{index}")
+    let indexes: Vec<_> = {
+        let mut indexes = store
+            .indexes_for_entity(&deployment_locator, entity_name)
+            .await?;
+        if no_attribute_indexes {
+            indexes = indexes
+                .into_iter()
+                .filter(|idx| !idx.is_attribute_index())
+                .collect();
+        }
+        if no_default_indexes {
+            indexes = indexes
+                .into_iter()
+                .filter(|idx| !idx.is_default_index())
+                .collect();
+        }
+        indexes
+    };
+
+    let mut term = Terminal::new();
+
+    if to_sql {
+        for index in indexes {
+            writeln!(term, "{};", index.to_sql(concurrent, if_not_exists)?)?;
+        }
+    } else {
+        let mut first = true;
+        header(&mut term, &indexes, &deployment_locator, entity_name)?;
+        for index in &indexes {
+            if !first {
```

Comment:
It seems this condition will never be satisfied, as `first` is never set to `false` outside that `if` expression.

```suggestion
            if first {
```

Reply:
Oh, yes - that was totally busted; fixing it makes the output a lot more readable